cfgov/alerts/send_new_relic_messages_to_sqs.py

```diff
@@ -67,52 +67,35 @@
 )
 
 
-def get_new_violations(newrelic_token, newrelic_url, threshold, policy_filter):
-    """ Check for violations in New Relic that are newer than the
-    newness_threshold in minutes. """
-    headers = {'X-Api-Key': newrelic_token}
-    violations_url = (newrelic_url +
-                      'alerts_violations.json?only_open=true')
-    r = requests.get(violations_url, headers=headers)
-    response_json = r.json()
-
-    violations = []
-    for violation in response_json['violations']:
-        logger.debug("Found violation: {violation}".format(
-                     violation=violation))
-        # Filter on the policy name
-        policy_name = violation['policy_name']
-        if policy_filter.search(policy_name) and \
-                violation['duration'] <= threshold:
-            violations.append(violation)
-
-    return violations
-
-
-def format_message_for_violation(violation, account_number):
-    """ Format the given violation dictionary into an SQS message
-    dictionary """
-    title = '{condition_name}, {entity_name}'.format(
-        condition_name=violation['condition_name'],
-        entity_name=violation['entity']['name']
-    )
-    incidents_link = (
-        'https://alerts.newrelic.com/accounts/'
-        '{account_number}/incidents'
-    ).format(
-        account_number=account_number
-    )
-    body = (
-        'New Relic {product}, {label}.'
-        '<a href="{link}">View incidents</a>'
-    ).format(
-        product=violation['entity']['product'],
-        type=violation['entity']['type'],
-        label=violation['label'],
-        link=incidents_link
-    )
-    message_body = '{title} - {body}'.format(title=title, body=body)
-    return message_body
+def cache_known_violations(known_violations_filename, known_violations):
```

Comment:
It seems like all this code around known violations could live inside `newrelic_alerts.py` instead. Then you could pass an instance of `NewRelicAlertViolations` the filename instead of the violations. Was there a reason for this separation?

Reply:
I thought maybe that file management was best left the script itself. That way we could use the same code in `newrelic_alerts` with some other management scheme for `know_violations` in the future. 

That might be overkill though. I kind of prefer it the way I've set it up, but moving the caching to `NewRelicAlertViolations` is certainly something I can do if you think it's necessary.